homeassistant/components/snapcast/media_player.py

```diff
@@ -275,6 +279,70 @@ async def async_unjoin(self) -> None:
         """Handle the unjoin service."""
         raise NotImplementedError
 
+    @property
+    def metadata(self) -> Mapping[str, Any]:
+        """Get metadata from the current stream."""
+        if metadata := self.coordinator.server.stream(
+            self._current_group.stream
+        ).metadata:
+            return metadata
+
+        # Fallback to an empty dict
+        return {}
+
+    @property
+    def media_title(self) -> str | None:
+        """Title of current playing media."""
+        return self.metadata.get("title")
+
+    @property
+    def media_image_url(self) -> str | None:
+        """Image url of current playing media."""
+        return self.metadata.get("artUrl")
+
+    @property
+    def media_artist(self) -> str | None:
+        """Artist of current playing media, music track only."""
+        return self.metadata.get("artist", [None])[0]
+
+    @property
+    def media_album_name(self) -> str | None:
+        """Album name of current playing media, music track only."""
+        return self.metadata.get("album")
+
+    @property
+    def media_album_artist(self) -> str | None:
+        """Album artist of current playing media, music track only."""
+        return self.metadata.get("albumArtist", [None])[0]
```

Comment:
shouldn't we concatenate all the possible artists?

Reply:
Yeah that does make sense. Does the form "Artist 1, Artist 2" work?